iceoryx_hoofs/testing/include/iceoryx_hoofs/testing/mocks/error_handler_mock.hpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_TESTUTILS_ERROR_HANDLER_MOCK_HPP
+#define IOX_HOOFS_TESTUTILS_ERROR_HANDLER_MOCK_HPP
+
+#include "iceoryx_hoofs/error_handling/error_handling.hpp"
+
+#include "iceoryx_hoofs/testing/test.hpp"
+
+namespace iox
+{
+template <typename Error>
+using TypedHandlerFunction = std::function<void(const Error, const ErrorLevel)>;
+
+class ErrorHandlerMock : protected ErrorHandler
+{
+  public:
+    template <typename Error>
+    static cxx::GenericRAII setTemporaryErrorHandler(const TypedHandlerFunction<Error>& newHandler) noexcept;
```

Comment:
Instead of having to call a function, why not make this the ctor?

Reply:
That would be possible. However, using `setTemporaryErrorHandler()` in tests is more meaningful for me. I use a mock to do set a certain behaviour.

Reply:
For me it is the oposite. It's like with the `std::lock`. One creates the object and let it's dtor do everything.